lib/tests/streamlit/runtime/caching/common_cache_test.py

```diff
@@ -872,3 +873,24 @@ def thread_test():
 
         # The other thread should not have modified the main thread
         self.assertEqual(1, get_counter())
+
+
+@patch("streamlit.source_util._cached_pages", new=None)
+class WidgetReplayInteractionTest(InteractiveScriptTests):
+    def test_dynamic_widget_replay(self):
+        script = self.script_from_filename("cached_widget_replay_dynamic.py")
+
+        sr = script.run()
+        assert len(sr.get("checkbox")) == 1
```

Comment:
This `get` behavior could be counterintuitive, maybe it could be a good idea to have `get_all()` and `get` could return only the first element, or we could implement `.first()` to don't force final test writers to get element by indices.

Reply:
I'm pretty attached to `get` returning all of them rather than calling that something else and making `get` a special case. I have found myself wanting a few other convenience methods not provided by the returned list, so maybe I'll bite the bullet and make things return a wrapper class that provides them, which would let me include `.first`